pkg/cmd/secvulnMaven.go

```diff
@@ -5,33 +5,319 @@
+		}
+
+		// Get the groupId of this project from the main pom to use in the url
+		var groupId string
+		if groupId = getGroupId(project, mainPom); groupId == "" {
+			fmt.Printf("Unable to get groupId for artifact %s\n", project)
+			panic(nil)
+		}
+		// Get the version of this project from the main pom to use in the url
+		var version string
+		if version = getVersion(project, mainPom); version == "" {
+			fmt.Printf("Unable to get version for artifact %s\n", project)
+			panic(nil)
+		}
+
+		// Get the current pom for this project to use to create the stripped down pom for the pseudo maven project
+		var currentPom Pom
+		if project == "com.jcraft.jsch" {
+			// com.jcraft.jsch is an exception to the normal format
+			currentPom, err = readPomFromRepos("jsch", "com.jcraft", version)
+		} else {
+			currentPom, err = readPomFromRepos(project, groupId, version)
+		}
+		if err != nil || (currentPom.ArtifactId != project && currentPom.ArtifactId != "jsch") {
+			fmt.Printf("Could not find pom for artifact %s\n", project)
+			panic(err)
+		}
+
+		createPseudoMavenProject(currentPom)
+
+		fmt.Printf("- %s\n", project)
+
+	}
+
+}
+
+func readPomFromUrl(url string) (Pom, error) {
+
+	var pom Pom
+
+	response, err := http.Get(url)
+	if err != nil {
+		return pom, err
+	}
+
+	defer response.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Body)
+	if err != nil {
+		return pom, err
+	}
+
+	err = xml.Unmarshal(body, &pom)
+	if err != nil {
+		return pom, err
+	}
+
+	return pom, nil
+
+}
+
+func readPomFromRepos(artifactName, groupId, version string) (Pom, error) {
+
+	var pom Pom
+	var err error
+
+	var url string
+	// Iterate through the provided repos with the --repo tag
+	for _, repo := range *secvulnMavenPomRepos {
+		// Use the groupId, artifactName and version to build up the url of the pom
+		s := strings.Split(groupId, ".")
+		var group string
+		for _, e := range s {
+			group += e + "/"
+		}
+
+		url = fmt.Sprintf("%s/%s%s/%s/%s-%s.pom", repo, group, artifactName, version, artifactName, version)
+
+		pom, err = readPomFromUrl(url)
+		if pom.ArtifactId == artifactName {
+			return pom, nil
+		}
+	}
+
+	return pom, err
+
+}
+
+func createPseudoMavenProject(pom Pom) {
+
+	var artifactName string
+	// com.jcraft.jsch is an exception so need to make sure it is in the correct format
+	if artifactName = pom.ArtifactId; artifactName == "jsch" {
+		artifactName = "com.jcraft.jsch"
+	}
+	createDirectory(artifactName)
+
+	// Using the current pom from the repo, create a stripped down pom for the pseudo maven project
+	createPom(pom, artifactName)
+
+	// Add this project to the list of completed projects so we don't reprocess and duplicate
+	completedProjects = append(completedProjects, artifactName)
+}
+
+func createDirectory(artifactName string) {
+	if err := os.Mkdir(fmt.Sprintf("%s/%s", secvulnMavenParentDir, artifactName), os.ModePerm); err != nil {
+		fmt.Printf("Unable to create directory for artifact %s - %v\n", artifactName, err)
```

Comment:
panic?

Reply:
done

Reply:
Actually, thinking of this, this would mean that if the directory is already there, this would fail and panic. Will the secvulnMavenParentDir that we provide to the galasabld command always be an empty directory (then panic would be fine), or should it just re-write over any directory that already exists (if so, then panic shouldn't be there)

Reply:
It will exist as we will clone the repo in the pipeline and replace in place.

Check to see if the directory exists before attempting to create it.